src/compiler/sys.ts

```diff
@@ -558,7 +558,14 @@ namespace ts {
                 directoryExists,
                 createDirectory(directoryName: string) {
                     if (!nodeSystem.directoryExists(directoryName)) {
-                        _fs.mkdirSync(directoryName);
+                        try {
+                            _fs.mkdirSync(directoryName);
+                        }
+                        catch (e) {
+                            if (e.code !== "EEXIST") {

```

Comment:
Do not check for this code, instead check if ```(!nodeSystem.directoryExists(directoryName))``` since EEXIST is also returned if file with same name exists.